src/SiteMaster/Core/Util.php

```diff
@@ -233,14 +233,23 @@ public static function getHTTPInfo($url, $options = array())
         
         curl_exec($curl);
 
-        $http_status = curl_getinfo($curl, CURLINFO_HTTP_CODE);
+        $http_status   = curl_getinfo($curl, CURLINFO_HTTP_CODE);
         $effective_url = curl_getinfo($curl, CURLINFO_EFFECTIVE_URL);
         $curl_error_no = curl_errno($curl);
 
         curl_close($curl);
         
+        //Always sleep after requesting data, this will help prevent loops from flooding a server.
+        sleep(1);
+        
         $okay = self::httpCodeIsOkay($http_status);
 
+        //Sometimes HEAD requests result in a false negative.  Do a GET request just to verify
+        if (true === $options[CURLOPT_NOBODY] && false === $okay) {
```

Comment:
Okay. We'll want to document this limitation (duplicate requests) somewhere other than the source file.

Also with the removal of the Guzzle client, care needs to be taken with regard to the cURL response. By default, cURL will output directly to `STDOUT` if an output file is not provided. Since you're not interested in the response body, perhaps sending it to `/dev/null` is ok.